R/derive_vars_cat.R

```diff
@@ -0,0 +1,218 @@
+#' Derive Categorization Variables Like `AVALCATy` and `AVALCAyN`
+#' @param dataset
+#' `r roxygen_param_dataset(expected_vars = c("by_vars", "definition"))`
+#' @param definition List of expressions created by: `exprs()`.
+#' Must be in the format of a tribble.
+#' Must contain:
+#'  - the column `condition` which evaluates to a logic in `dataset`.
+#'  - at least one additional column with the new column name and the category value.
+#'  - the column specified in `by_vars` (if `by_vars` is specified)
+#'
+#' e.g. if `by_vars` is not specified:
+#' `exprs(~condition, ~AVALCAT1, ~AVALCA1N
+#'        AVAL >= 140, ">=140 cm", 1,
+#'        AVAL < 140, "<140 cm", 2)`
+#'
+#' e.g. if `by_vars` is specified as `exprs(VSTEST)`:
+#' `exprs(~VSTEST, ~condition, ~AVALCAT1, ~AVALCA1N
+#'        "Height", AVAL >= 140, ">=140 cm", 1,
+#'        "Height", AVAL < 140, "<140 cm", 2)`
```

Comment:
Not sure if possible, but can we make this into a table so it is more readable 
![image](https://github.com/user-attachments/assets/835c91a9-61bf-4254-8138-e7e648e744fa)

Reply:
I feel like in the R file it is quite readable. 
Are you refering to how it is rendered in the manual?

Reply:
Done